app/controllers/facilities_controller.rb

```diff
@@ -75,12 +76,47 @@ def manage
     @active_tab = "admin_facility"
   end
 
+  require "zip"
+
+  #Start
+
+  # GET /facilities/:facility_id/thumbnail
+  def thumbnail
+    raise ActiveRecord::RecordNotFound if @facility.thumbnail.template.empty?
+    @file = @facility.thumbnail.new(file_type: "img")
+  end
+
+  # POST /facilities/:facility_id/upload_thumbnail
+  def upload_thumbnail
```

Comment:
Can we be a little more RESTful and create a new controller?  It would just have `create` and `destroy` actions. Our normal pattern would be `FacilityThumbnailsController`. I don't love the pattern, but it's what we've got.

Reply:
Or are these actions even used? It looks like it's going through the normal `update`.